File: src/2025-03-04-03-28-40--deepseek-chat-0304/go/exercises/practice/bottle-song/bottle_song.go
package bottlesong

import (
	"fmt"
)

func Recite(startBottles, takeDown int) []string {
	var verses []string
	for i := startBottles; i > startBottles-takeDown; i-- {
		if i == 0 {
			break
		}
		
		// First two lines
		verse := fmt.Sprintf("%s green %s hanging on the wall,", capitalize(numberToWord(i)), pluralize(i))
		verses = append(verses, verse, verse)
		
		// Third line
		verses = append(verses, "And if one green bottle should accidentally fall,")
		
		// Fourth line
		remaining := i - 1
		if remaining == 0 {
			verses = append(verses, "There'll be no green bottles hanging on the wall.")
		} else {
			verses = append(verses, fmt.Sprintf("There'll be %s green %s hanging on the wall.", numberToWord(remaining), pluralize(remaining)))
		}
		
		// Add empty line between verses except the last one
		if i > startBottles-takeDown+1 {
			verses = append(verses, "")
		}
	}
	return verses
}

func numberToWord(n int) string {
	words := []string{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"}
	if n >= 0 && n <= 10 {
		return words[n]
	}
	return ""
}

func capitalize(s string) string {
	if len(s) == 0 {
		return s
	}
	return string(s[0]-32) + s[1:]
}

func pluralize(n int) string {
	if n == 1 {
		return "bottle"
	}
	return "bottles"
}

File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/bottle-song/bottle-song.js
const numberToWords = (num) => {
  const words = [
    'Zero', 'One', 'Two', 'Three', 'Four', 
    'Five', 'Six', 'Seven', 'Eight', 'Nine', 'Ten'
  ];
  return words[num];
};

const getVerse = (bottles) => {
  if (bottles === 0) {
    return ['No green bottles hanging on the wall.'];
  }
  
  const bottleText = bottles === 1 ? 'bottle' : 'bottles';
  const nextBottles = bottles - 1;
  const nextBottleText = nextBottles === 1 ? 'bottle' : 'bottles';
  
  const currentWord = numberToWords(bottles);
  const nextWord = numberToWords(nextBottles);
  
  return [
    `${currentWord} green ${bottleText} hanging on the wall,`,
    `${currentWord} green ${bottleText} hanging on the wall,`,
    `And if one green bottle should accidentally fall,`,
    nextBottles > 0 
      ? `There'll be ${nextWord.toLowerCase()} green ${nextBottleText} hanging on the wall.`
      : `There'll be no green bottles hanging on the wall.`
  ];
};

export const recite = (initialBottlesCount, takeDownCount) => {
  let result = [];
  let currentBottles = initialBottlesCount;
  
  for (let i = 0; i < takeDownCount; i++) {
    result.push(...getVerse(currentBottles));
    currentBottles--;
    
    // Add an empty line between verses except the last one
    if (i < takeDownCount - 1) {
      result.push('');
    }
  }
  
  return result;
};
